content/docs/k8s/gateway-api.mdx

```diff
@@ -0,0 +1,178 @@
+
+### Pomerium-specific extensions
+
+Pomerium defines a PolicyFilter CRD for setting the authorization policy for a particular route. Policies are expressed using [Pomerium Policy Language](/docs/capabilities/ppl) (PPL).
+
+Pomerium will deny access to any route without an explicit authorization, so you must set a PolicyFilter on every route you define for use with Pomerium.
+
+## Installation
+
+To install the Pomerium Ingress Controller and enable Gateway API support:
+
+1. First install the Gateway API CRDs:
+
+   ```sh
+   kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.2.0/standard-install.yaml
+   ```
+
+1. Then install the Pomerium Ingress Controller:
+
+   ```sh
+   kubectl apply -k github.com/pomerium/ingress-controller/config/gateway-api\?ref=v0.28.0
+   ```
+
+## Example usage
+
+Let's look at how you can configure a route and policy using the Gateway API.
+
+1. We'll need a TLS certificate for the domain to be used with this route. (For testing purposes you could generate a certificate using [mkcert].)
+
+   For this example we'll assume we have a wildcard certificate for `*.localhost.pomerium.io`, and we'll create a Secret to store this certificate:
+
+   ```bash
+   kubectl create secret tls pomerium-wildcard-tls \
+     --cert=_wildcard.localhost.pomerium.io.pem \
+     --key=_wildcard.localhost.pomerium.io-key.pem
+   ```
+
+1. Next we'll set up a Gateway that uses this certificate:
+
+   ```yaml title="pomerium-gateway.yaml"
+   apiVersion: gateway.networking.k8s.io/v1
+   kind: Gateway
+   metadata:
+     name: localhost-pomerium-io
+   spec:
+     gatewayClassName: pomerium-gateway
+     listeners:
+       - name: https
+         hostname: '*.localhost.pomerium.io'
+         protocol: HTTPS
+         port: 443
+         tls:
+           mode: Terminate
+           certificateRefs:
+             - name: pomerium-wildcard-tls
+   ```
+
+   Deploy with `kubectl apply -f pomerium-gateway.yaml`.
+
+1. We'll also need a service. Here we'll use httpbin as an example:
+
+   ```yaml title="httpbin-service.yaml"
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: httpbin
+   spec:
+     selector:
+       app: httpbin
+     ports:
+       - port: 80
+         targetPort: 80
+         name: http
+   ---
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     name: httpbin-deployment
+     labels:
+       app: httpbin
+   spec:
+     replicas: 1
+     selector:
+       matchLabels:
+         app: httpbin
+     template:
+       metadata:
+         labels:
+           app: httpbin
+       spec:
+         containers:
+           - name: httpbin
+             image: docker.io/kennethreitz/httpbin
+             ports:
+               - containerPort: 80
+   ```
+
+   Deploy with `kubectl apply -f httpbin-service.yaml`.
```

Comment:
We do use the verify app in the main quickstart, but in this initial Gateway API implementation we don't yet have a filter for adding the Pomerium JWT to a request, so I figured it didn't make sense to show the verify app in this example yet.

Once we can show this working end-to-end it might be good to switch over to the verify app here.